pm/sendermonitor_test.go

```diff
@@ -380,6 +424,51 @@ func TestCleanup(t *testing.T) {
 	mf2, err = sm.MaxFloat(addr2)
 	require.Nil(err)
 
-	assert.Equal(reserve4, mf1)
-	assert.Equal(reserve5, mf2)
+	expectedAlloc4 := new(big.Int).Sub(new(big.Int).Div(smgr.info[addr2].Reserve, rm.transcoderPoolSize), smgr.claimedReserve[addr2])
+	assert.Equal(expectedAlloc3, mf1)
+	assert.Equal(expectedAlloc4, mf2)
+}
+
+func TestReserveAlloc(t *testing.T) {
+	assert := assert.New(t)
+
+	claimant := RandAddress()
+	b := newStubBroker()
+	addr := RandAddress()
+	smgr := newStubSenderManager()
+	smgr.info[addr] = &SenderInfo{
+		Deposit:       big.NewInt(500),
+		Reserve:       big.NewInt(5000),
+		WithdrawBlock: big.NewInt(0),
+		ReserveState:  NotFrozen,
+		ThawRound:     big.NewInt(0),
+	}
+	smgr.claimedReserve[addr] = big.NewInt(100)
+	rm := &stubRoundsManager{
+		transcoderPoolSize: big.NewInt(5),
+	}
+	em := &stubErrorMonitor{}
+	sm := &senderMonitor{
```

Comment:
Can we just use the `NewSenderMonitor` constructor here?

Reply:
We can't because NewSenderMonitor returns the interface, not the concrete type. Therefore we wouldn't have access to the unexported `reserveAlloc` method which we are testing in this test.

Reply:
Ah right. A shortcut here could be to use `NewSenderMonitor` and then use a type assertion to convert the interface into the `senderMonitor` concrete type.